core/src/main/java/io/neow3j/protocol/core/methods/response/StackItem.java

```diff
@@ -0,0 +1,187 @@
+            return (ByteArrayStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.BYTE_ARRAY.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public BooleanStackItem asBoolean() {
+        if (this instanceof BooleanStackItem) {
+            return (BooleanStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.BOOLEAN.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public IntegerStackItem asInteger() {
+        if (this instanceof IntegerStackItem) {
+            return (IntegerStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.INTEGER.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public ArrayStackItem asArray() {
+        if (this instanceof ArrayStackItem) {
+            return (ArrayStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.ARRAY.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public MapStackItem asMap() {
+        if (this instanceof MapStackItem) {
+            return (MapStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.MAP.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public StructStackItem asStruct() {
+        if (this instanceof StructStackItem) {
+            return (StructStackItem) this;
+        }
+        throw new IllegalStateException("This stack item is not of type " +
+                StackItemType.STRUCT.jsonValue() + " but of " + this.type.jsonValue());
+    }
+
+    public static class StackDeserializer extends StdDeserializer<StackItem> {
+
+        protected StackDeserializer() {
+            this(null);
+        }
+
+        protected StackDeserializer(Class<StackItem> vc) {
+            super(vc);
+        }
+
+        public StackItem deserialize(JsonParser jp, DeserializationContext ctxt)
+                throws IOException {
+
+            JsonNode node = jp.getCodec().readTree(jp);
+            return deserializeStackItem(node, jp);
+        }
+
+        private StackItem deserializeStackItem(JsonNode itemNode, JsonParser jp)
+                throws JsonProcessingException {
+
+            JsonNode typeNode = itemNode.get("type");
+            JsonNode valueNode = itemNode.get("value");
+            StackItemType type = null;
+            if (typeNode != null) {
+                type = jp.getCodec().treeToValue(typeNode, StackItemType.class);
+            }
+            if (valueNode == null) {
+                return new StackItem(type, null);
+            }
+            if (type == null) {
+                return new StackItem(null, valueNode.asText());
+            }
+            switch (type) {
```

Comment:
Yeah, this is a bit nasty, but we can live with it... I try to not use deal with JSON serialization/deserialization as much as possible, but sometimes we can't avoid to use it...